include/lug/System/Memory.hpp

```diff
@@ -0,0 +1,145 @@
+#pragma once
+
+#include <cstddef>
+#include <type_traits>
+#include <new>
+#include <memory>
+
+#include <lug/System/Memory/Arena.hpp>
+#include <lug/System/Memory/Policies/Thread.hpp>
+#include <lug/System/Memory/Policies/BoundsChecker.hpp>
+#include <lug/System/Memory/Policies/MemoryMarker.hpp>
+
+
+// The variadic arguments are always "arena, args..." (args only work for non array allocation and array of non pod)
+#define LUG_NEW_ALIGN(T, alignment, ...) ::lug::System::Memory::new_one<T>(alignment, __FILE__, __LINE__, __VA_ARGS__)
+#define LUG_NEW_ARRAY_ALIGN(T, alignment, ...) LUG_NEW_ARRAY_ALIGN_SIZE(typename std::remove_all_extents<T>::type, alignment, std::extent<T>::value, __VA_ARGS__)
+#define LUG_NEW_ARRAY_ALIGN_SIZE(T, alignment, size, ...) ::lug::System::Memory::new_array<typename std::remove_all_extents<T>::type>(alignment, size, __FILE__, __LINE__, __VA_ARGS__)
+
+#define LUG_NEW(T, ...) LUG_NEW_ALIGN(T, alignof(T), __VA_ARGS__)
+#define LUG_DELETE(object, arena) ::lug::System::Memory::delete_one(object, arena)
+
+#define LUG_NEW_ARRAY(T, ...) LUG_NEW_ARRAY_ALIGN(T, alignof(typename std::remove_all_extents<T>::type), __VA_ARGS__)
+#define LUG_NEW_ARRAY_SIZE(T, size, ...) LUG_NEW_ARRAY_ALIGN_SIZE(T, alignof(typename std::remove_all_extents<T>::type), size, __VA_ARGS__)
+#define LUG_DELETE_ARRAY(object, arena) ::lug::System::Memory::delete_array(object, arena)
```

Comment:
I have explain that in Slack, it's just in case if the macro is used in some namespace context we just use the namespace from the "root" namespace.